### PR TITLE
portal alteration importer mapping fix

### PR DESCRIPTION
**Pull request by JiaoJiao123**
There are two places fixed.
a) proteinPosStart should be used instead of startPosition. Same thing for the endPosition. This caused the portal alterations failed to fall into the oncokb alteration protein range.
b) use entrezGeneId or hugoSymbol to compare gene instead of object

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java [hunk 1]
```diff
@@ -60,7 +60,8 @@ public Alteration findAlteration(Gene gene, AlterationType alterationType, Strin
         if (gene != null && consequence != null && !consequence.getTerm().equals("NA")) {
             if (alterations != null && alterations.size() > 0) {
                 for (Alteration alteration : alterations) {
-                    if (alteration.getGene().equals(gene) && alteration.getConsequence() != null && alteration.getConsequence().equals(consequence)) {
+                    if ((alteration.getGene().getEntrezGeneId() != null && alteration.getGene().getEntrezGeneId().equals(gene.getEntrezGeneId()) || alteration.getGene().getHugoSymbol() != null && alteration.getGene().getHugoSymbol().equals(gene.getHugoSymbol()))
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java]: Why changed the logic?

**Comment by JiaoJiao123** [comment]
@zhx828 In the PortalAlterationImporter.java, Line 120 and Line 121 are modified. The rest are unchanged.

File: core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java (hunk 1 above)
Comment (JiaoJiao123) [reply core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java]: The original logic was "alteration.getGene().equals(gene)" and it kept returning false. This is the problem I asked you earlier. As I understand, both gene and alteration came from the same cache,  I couldn't figure out why they are different. Sorry it is bad practice to change things I don't understand to just make it work. I will ask you more about this issue tomorrow.

**Comment by zhx828** [comment]
This solves issue https://github.com/oncokb/oncokb/issues/878

**Pull request closed without merging**